native/src/main/java/io/ballerina/stdlib/graphql/runtime/engine/ListenerUtils.java

```diff
@@ -128,12 +128,18 @@ public static BString sanitizeBasePath(String basePath) {
         return StringUtils.fromString(basePath.trim());
     }
 
-    public static Object getHtmlContentFromResources(BString url) {
+    public static Object getHtmlContentFromResources(BString url, Object subscriptionUrl) {
         InputStream htmlAsStream = ClassLoader.getSystemResourceAsStream(GRAPHIQL_RESOURCE);
         try {
             byte[] bytes = htmlAsStream.readAllBytes();
             String htmlAsString = new String(bytes, StandardCharsets.UTF_8);
-            htmlAsString = htmlAsString.replace(REGEX_URL, url.getValue());
+            StringBuilder graphiqlUrl = new StringBuilder("{ url: \"" + url.getValue() + "\" }");
+            if (subscriptionUrl != null) {
+                graphiqlUrl.deleteCharAt(graphiqlUrl.length() - 1);
```

Comment:
Instead of adding and deleting the character, you can add the close brace at the end.